### PR TITLE
New how-tos

### PR DESCRIPTION
**Pull request by Astarsys**

### CODE REVIEW

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+We need to create a status listing that we will assign to the device. Based on the statuses that are created for the sheet
+- we will have a choice - what status can be assigned to the tracker.
+
+To create the listing we need only one parameter:
+* `listing` - [status_listing](../../status/index.md#status-listing-object-structure) object without "id" and "entries" fields.
+
+For example, we will create a listing for the delivery service to allow drivers and supervisors to change the status.
+Drivers can change their status using the X-GPS app. Supervisors can change status using the UI.
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:23]: Add internal reference to API resource `/status/listing/create`.

File: docs/backend-api/how-to/how-to-send-commands-to-device.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+`tracker_id` - ID of the device to which we want to send the command.
+`command` - Text or hexadecimal representation of the command in a protocol-dependent manner.
+
+For example, we have a Teltonika FMB140 device with ID on the platform 231402. And I want to send a command to reconfigure
+its IP address to a new one 52.57.1.136. According to the protocol the command should be the next:
+
+`setparam 2004:52.57.1.136`
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-send-commands-to-device.md:27]: Add internal reference to API resource `tracker/raw_command/send`.

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    "id": 1111
+}
+```
+
+When we created a listing, we need to fill it with statuses, and we should use one request per one status.
+
+For example, we have 4 statuses:
+"Free," "Break," "Pick up the goods from storage," "Deliver goods."
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:47]: Add internal reference to API resource `/status/create`.

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+}
+```
+
+## Assign
+
+To assign the listing to some devices, we use the `tracker_id` and `listing_id.`
+
+For example, we have 10 drivers. We should create 10 requests for assigning the listing to them.
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:72]: Add internal reference to API resource `/status/listing/tracker/assign`.

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+=== "cURL"
+
+    ```shell
+    curl -X POST '{{ extra.api_example_url }}/status/listing/tracker/assign' \
+        -H 'Content-Type: application/json' \ 
+        -d '{"hash": "22eac1c27af4be7b9d04da2ce1af111b", "tracker_id": "615487", "listing_id": "111"}'
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:79]: `listing_id` and `tracker_id` is not a string

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+"Free," "Break," "Pick up the goods from storage," "Deliver goods."
+
+API request:
+
+=== "cURL"
+
+    ```shell
+    curl -X POST '{{ extra.api_example_url }}/status/create' \
+        -H 'Content-Type: application/json' \ 
+        -d '{"hash": "22eac1c27af4be7b9d04da2ce1af111b", "listing_id": "1111", "status": {"label": "Free", "color": "E57373"}}'
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:54]: `listing_id` is not a string

File: docs/backend-api/how-to/how-to-send-commands-to-device.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+=== "cURL"
+
+    ```shell
+    curl -X POST '{{ extra.api_example_url }}/tracker/raw_command/send' \
+        -H 'Content-Type: application/json' \
+        -d '{"hash": "a6aa75587e5c59c32d347da438505fc3", "tracker_id": "231402", "command": "setparam 2004:52.57.1.136"}'
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-send-commands-to-device.md:34]: `tracker_id` is not a string

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    curl -X POST '{{ extra.api_example_url }}/status/listing/tracker/assign' \
+        -H 'Content-Type: application/json' \ 
+        -d '{"hash": "22eac1c27af4be7b9d04da2ce1af111b", "tracker_id": "615487", "listing_id": "111"}'
+    ```
+
+The platform will notify you about success in reply.
+
+After that, our drivers and supervisors will have access to change statuses. Where they could be used:
+We can create a script that will assign a new task to a driver with the status "Free." After assigning the task, this script
+will change the status itself to "Pick up the goods from storage."
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:86]: A dot at the end should be out of quotes.

File: docs/backend-api/how-to/how-to-work-with-statuses.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    ```shell
+    curl -X POST '{{ extra.api_example_url }}/status/listing/tracker/assign' \
+        -H 'Content-Type: application/json' \ 
+        -d '{"hash": "22eac1c27af4be7b9d04da2ce1af111b", "tracker_id": "615487", "listing_id": "111"}'
+    ```
+
+The platform will notify you about success in reply.
+
+After that, our drivers and supervisors will have access to change statuses. Where they could be used:
+We can create a script that will assign a new task to a driver with the status "Free." After assigning the task, this script
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-statuses.md:85]: A dot at the end should be out of quotes.

File: docs/backend-api/how-to/how-to-work-with-tasks.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+* `tracker_id` - to which tracker this task should be assigned.
+* `location` - where the task should be.
+* `label` - the name of a new task.
+* `description` - a note about the task.
+* `from` and `to` - when the task should be completed.
+* `max_delay` - the employee may be late with the execution for a maximum of this time in minutes.
+* `min_stay_duration` - the task will not be considered completed if the employee spends less than this time in the task's zone.
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-tasks.md:40]: Add internal reference to API resource `/task/create`.

File: docs/backend-api/how-to/how-to-work-with-tasks.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+The list of necessary parameters is next:
+
+* `route` - JSON object containing all necessary information about the [route](../resources/field_service/task/route/index.md#route-object) without *IGNORED* fields.
+* `checkpoint` - array of [checkpoint objects](../resources/field_service/task/checkpoint.md#checkpoint-object) without *IGNORED* fields.
+* `create_form` - boolean. If `true` then check additional `form_template_id` field in every **checkpoint** object and create form if it is not null. Default value is `false` for backward compatibility.
+
+For example, we need to create the next route:
+
+John needs to deliver our products to three customers on 18th of March, from 10 AM to 4 PM. His car has a tracker with
+id 120945. He can't get late because our customers will wait for production at the exact time and if he is late - the checkpoint
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-tasks.md:72]: Should this tracker id used in the cURL example?

File: docs/backend-api/how-to/how-to-work-with-tasks.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+to hand over the goods to the client and fill out the documents.
+
+In this case, every `checkpoint` object will have the next parameters:
+
+* `tracker_id` - an id of the tracker to which checkpoint should be assigned.
+* `location` - location associated with this checkpoint. cannot be null.
+* `label` - the name of the checkpoint.
+* `description` - a note about the checkpoint.
+* `from` and `to` - the time when this checkpoint should be completed.
+* `external_id` - this is a delivery code. It is necessary for a checkpoint because I have the plugin "Courier on the map."
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-tasks.md:83]: A dot at the end should be out of quotes.

File: docs/backend-api/how-to/how-to-work-with-tasks.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+  of delivery, photos of delivered products, bill, and customer's signature.
+  
+The route object will have its own parameters too:
+
+* `tracker_id` - an id of the tracker to which a route should be assigned.
+* `label` - route name.
+* `description` - additional information about the whole route.
+* `from` and `to` - the time when this route should be completed.
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-tasks.md:99]: Add internal reference to API resource `/task/route/create`.

File: docs/backend-api/how-to/how-to-work-with-tasks.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+before route creation. Our platform provides a way to perform such optimization. Provide data required to optimize, and the 
+algorithm returns order in which points should be visited. Specify checkpoint objects in this order when you will create
+the route.
+
+Necessary parameters:
+
+* `start_point` - JSON object with the point and time from that our driver will start the move.
+* `route_points` - an array of JSON objects with points and time that driver should visit.
+
+API request:
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to-work-with-tasks.md:149]: Add internal reference to API resource `/task/route/points/optimize`.

File: docs/backend-api/how-to/how-to.md [hunk 15]
```diff
@@ -19,4 +19,7 @@ needs but also help you make your business more profitable.
 * [How to get track points](./get-track-points.md)
 * [How to obtain report's information](./how-to-obtain-information-from-report.md)
 * [How to use rules](./how-to-create-a-rule.md)
-* [How to work with notifications](./how-to-work-with-notifications.md)
\ No newline at end of file
+* [How to work with notifications](./how-to-work-with-notifications.md)
+* [How to create, assign and read status of tasks](./how-to-work-with-tasks.md)
```
Comment (fenik17) [inline docs/backend-api/how-to/how-to.md:23]: Maybe: "How to create/assign the tasks and optimize route"?

**Review by fenik17 — CHANGES_REQUESTED** [review]

**Comment by Astarsys** [comment]
ready

**Review by fenik17 — APPROVED** [review]

**Pull request merged**